src/runtime/arch_avr.go

```diff
@@ -14,3 +17,14 @@ func align(ptr uintptr) uintptr {
 }
 
 func getCurrentStackPointer() uintptr
+
+//export __sync_val_compare_and_swap_2
+func __sync_val_compare_and_swap_2(ptr *uint32, expected, desired uint32) uint32 {
```

Comment:
The same goes for AVR.

Note that atomics are somewhat broken on AVR. If you see weird behavior on AVR, it could be the backend that is emitting invalid instructions.
For more details, see: https://reviews.llvm.org/D97127